packages/dev/core/src/Meshes/abstractMesh.ts

```diff
@@ -822,13 +823,20 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
     constructor(name: string, scene: Nullable<Scene> = null) {
         super(name, scene, false);
 
-        this.getScene().addMesh(this);
+        scene = this.getScene();
+
+        scene.addMesh(this);
 
         this._resyncLightSources();
 
         // Mesh Uniform Buffer.
         this._uniformBuffer = new UniformBuffer(this.getScene().getEngine(), undefined, undefined, name, !this.getScene().getEngine().isWebGPU);
         this._buildUniformLayout();
+
+        if (scene.performancePriority !== ScenePerformancePriority.BackwardCompatible) {
+            this.alwaysSelectAsActiveMesh = true;

```

Comment:
Do we really want to do that?

I think there could be cases where setting `alwaysSelectAsActiveMesh=true` will actually make perf worse...

Maybe do it only in the **Agressive** mode?

[EDIT] Doing it only in the **Agressive** mode would be moot as there's already `this._skipFrustumClipping=true` in this mode...

Reply:
We are massively CPU bound so I truly believe this is required